### PR TITLE
rust: remove `Arc` from rust samples.

### PR DESCRIPTION
**Pull request by wedsonaf**
Signed-off-by: Wedson Almeida Filho <wedsonaf@google.com>

### CODE REVIEW

**Comment by ksquirrel** [comment]
Review of `1da768be0fd7`:
  - ✔️ Commit [`1da768b`](https://github.com/Rust-for-Linux/linux/commit/1da768be0fd7ce4dd4fafa7a42cfe0cee1ef020e): Looks fine!

File: samples/rust/rust_semaphore.rs [hunk 1]
```diff
@@ -111,31 +111,34 @@ impl FileOperations for FileState {
...
+                    })
+                },
             },
-        })?;
-
-        // SAFETY: `changed` is pinned behind `Arc`.
-        condvar_init!(Pin::new_unchecked(&sema.changed), "Semaphore::changed");
+            |sema| {
+                // SAFETY: `changed` is pinned when `sema` is.
+                condvar_init!(Pin::new_unchecked(&sema.changed), "Semaphore::changed");
```
Comment (TheSven73) [inline samples/rust/rust_semaphore.rs:136]: Shouldn't this be
```rust
condvar_init!(unsafe { Pin::new_unchecked(&sema.changed) }, "Semaphore::changed");
```
instead? Why is the missing `unsafe` not enforced?

That said, if this is indeed a problem, it should likely go in its own PR. So this comment shouldn't hold up the PR.

**Review by TheSven73 — APPROVED** [review]
LGTM

File: samples/rust/rust_semaphore.rs (hunk 1 above)
Comment (ojeda) [reply samples/rust/rust_semaphore.rs:136]: It is because `init_with_lockdep!` wraps the expression inside `unsafe`.

Comment (wedsonaf) [reply samples/rust/rust_semaphore.rs:136]: Nice catch, Sven! Uploaded #391 to fix this.
